project/src/pages/favorites/favorites.tsx

```diff
@@ -48,7 +40,7 @@ function Favorites(offers: FavoritesProps): JSX.Element {
           <section className="favorites">
             <h1 className="favorites__title">Saved listing</h1>
             <ul className="favorites__list">
-              {groupOffers.map((group) => <FavoritesItem className="favorites__locations-items" group={group} key={group.city}/>)}
+              {Object.entries(groupOffers).map((group: OffersGroup) => <FavoritesItem className="favorites__locations-items" group={group} key={offers.length} onMouseOver={onMouseOver} />)}
```

Comment:
Зачем вводить новый тип, если это просто производная и можно обойтись без нее. Тс знает за счет типизации редьюса объект какого вида здесь
```suggestion
              {Object.entries(groupOffers).map(([city, offers]) => <FavoritesItem className="favorites__locations-items" city={city} offers={offers} key={offers.length} onMouseOver={onMouseOver} />)}
```
Почему key который должен быть уникальным зависит от длины массива?
FavoritesItem перенести на след строку и все ключи столбиком